stock_picking_actual_date/models/stock_move.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023-2024 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    actual_date = fields.Date(
+        compute="_compute_actual_date",
+        store=True,
+    )
+
+    @api.depends("date", "picking_id.actual_date")
+    def _compute_actual_date(self):
+        for rec in self:
+            if rec.picking_id.actual_date:
+                rec.actual_date = rec.picking_id.actual_date
+                continue
+            rec.actual_date = fields.Datetime.context_timestamp(self, rec.date)
```

Comment:
Can we take the timezone from the company?

Reply:
I made the adjustment. But for the existing records values that come from migration script will not effect the conversion of timezone. We can add the schedule action to run compute function for all records if we need to handle.